src/Mod/Mesh/App/Exporter.cpp

```diff
@@ -54,7 +56,32 @@ std::string Exporter::xmlEscape(const std::string &input)
     return out;
 }
 
-MergeExporter::MergeExporter(std::string fileName, MeshIO::Format fmt)
+bool Exporter::addAppGroup(App::DocumentObject *obj, float tol)
+{
+    const auto meshFeatId( Base::Type::fromName("Mesh::Feature") );
+    const auto partFeatId( Base::Type::fromName("Part::Feature") );
+    const auto appPartId( Base::Type::fromName("App::Part") );
+    const auto appDOGId( Base::Type::fromName("App::DocumentObjectGroup") );
+
+    auto ret(true);
+
+    for (auto it : static_cast<App::Part *>(obj)->getOutList()) {
+        if (it->getTypeId().isDerivedFrom(meshFeatId)) {
+            ret &= addMeshFeat(it);
+        } else if (it->getTypeId().isDerivedFrom(partFeatId)) {
+            ret &= addPartFeat(it, tol);
+        } else if ( it->getTypeId().isDerivedFrom(appPartId) ||
```

Comment:
Parts and Groups are defined as Extension and hence cannot always be detected by only checking the DocumentObject type. For exampel your check would not find the PartDesign Body, even thought it behaves like a App::Part in many ways. I don't know your intend here, but if you like to get all kind of objects that behave like aApp::Part you need to check for the extension:
it->hasExtension(App::GeoFeatureGroupExtension::getExtensionClassId())

Note that the same holds for Group, wheer you check GroupExtension::getExtensionClassTypeId(). But be aware thet this returns tru efor Groups and GeofeatureGroups (as GeoFeatureGroup derives from Group)